base/gmp.jl

```diff
@@ -521,8 +521,11 @@ hex(n::BigInt) = base(16, n)
 for f in (:bin, :oct, :dec, :hex)
     @eval function ($f)(n::BigInt, pad::Int)
         b = IOBuffer()
-        res = ($f)(n)
+        res = ($f)(abs(n))
```

Comment:
This unfortunately creates another BigInt for `abs(n)`, which could be quite expensive. It would be preferable to use the GMP API to print `n` with padding without constructing `abs(n)`. One way to do it would be to temporarily construct `abs(n)` while sharing the same data pointer as `n`, but that may be a bit too tricky. Using GMP printing functions directly is probably the better approach.